packages/neos-ui-extensibility/src/registry/SynchronousRegistry.js

```diff
@@ -1,38 +1,74 @@
 import AbstractRegistry from './AbstractRegistry';
+import {positionalArraySorter} from '@neos-project/utils-helpers';
 
 export default class SynchronousRegistry extends AbstractRegistry {
     constructor(description) {
         super(description);
 
-        // internal registry, containing the keys as object keys, and the values being the values in the registry.
-        this._registry = {};
-        this._keys = [];
+        this._registry = [];
     }
 
-    add(key, value) {
-        // TODO: implement position in registry!
-        // TODO: "key" must be string!
-        this._registry[key] = value;
-        this._keys.push(key);
+    set(key, value, position = 0) {
```

Comment:
I am not sure we should allow having the same key twice. I think this should always be an error, as it makes the remaining API inconsistent / indeterminate

Reply:
Great point!

Reply:
On the other hand, what about if the user wants to override something in our UI? E.g. replace the tree with smth else?